src/main/java/io/stargate/sgv2/jsonapi/service/processor/MeteredCommandProcessor.java

```diff
@@ -258,10 +259,14 @@ private <T extends SchemaObject> Tags getCustomTags(
     return tags;
   }
 
-  private JsonApiMetricsConfig.SortType getVectorTypeTag(Command command) {
+  private JsonApiMetricsConfig.SortType getVectorTypeTag(
+      CommandContext<?> commandContext, Command command) {
     int filterCount = 0;
-    if (command instanceof Filterable fc && fc.filterClause() != null) {
-      filterCount = fc.filterClause().logicalExpression().getTotalComparisonExpressionCount();
+    if (command instanceof Filterable filterable) {
+      FilterClause fc = filterable.filterClause(commandContext);
+      if (fc != null) {
```

Comment:
these are the types of situations I think it would be good to make Filterable.filterClause always return non-null

Reply:
Yes, I was toying with the idea of adding "empty" `FilterClause` always. That's a possibility -- unless we want to know difference between missing and empty clause? (not sure we do; probably we don't).